docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -0,0 +1,104 @@
+---
+
+# LOOKUP JOIN [esql-lookup-join]
+
+The {{esql}} [`LOOKUP join`](/reference/query-languages/esql/esql-commands.md#esql-lookup-join) processing command combines, at query-time, data from one or more source indexes with field-value combinations found in an input table. Teams often have data scattered across multiple indices – like logs, IPs, user IDs, hosts, employees etc. Without a direct way to enrich or correlate each event with reference data, root-cause analysis, security checks, and operational insights become time-consuming.
+
+For example, you can use `LOOKUP JOIN` to:
+
+* Pull in environment or ownership details for each host to correlate your metrics data.
+* Quickly see if any source IPs match known malicious addresses.
+* Tag logs with the owning team or escalation info for faster triage and incident response.
+
+[`LOOKUP join`](/reference/query-languages/esql/esql-commands.md#esql-lookup-join) is similar to [`ENRICH`](/reference/query-languages/esql/esql-commands.md#esql-enrich) in the fact that they both help you join data together. You should use `LOOKUP JOIN` when:
+
+* Enrichment data changes frequently
+* You want to avoid index time processing
+* Working with regular indices
+* Need to preserve distinct matches
+* Need to match on any field in a lookup index
+
+## How the `LOOKUP JOIN` command works [esql-how-lookup-join-works]
+
+The `LOOKUP JOIN` command adds new columns to a table, with data from {{es}} indices.
+
+:::{image} ../../../images/esql-lookup-join.png
+:alt: esql lookup join
+:::
+
+$$$esql-source-index$$$
+
+Source index
+:   An index which stores enrich data that the `LOOKUP` command can add to input tables. You can create and manage these indices just like a regular {{es}} index. You can use multiple source indices in an enrich policy. You also can use the same source index in multiple enrich policies.
+
+## Example
+
+In the case where there are multiple matches on the index `LOOKUP JOIN` the output rows is the combination of each match from the left with each match on the right. 
+
+Image you have the two tables:
+
+**Left**
+
+|Key|Entry|
+| --- | --- |
+|1|A|
+|2|B|
+
+**Right**
+
+|Key|Language|TLD|
+|---|---|---|
+|1|English|CA|
+|1|English|null|
+|1|null|UK|
+|1|English|US|
+|2|German|AT,DE|
+|2|German|CH|
+|2|German|null|
+
+Running the following query would provide the results shown below.
+
+```esql
+FROM Left
+| WHERE Language IS NOT NULL // works and filter TLD UK
+| LOOKUP JOIN Right ON Key
+```
+
+|Key|Entry|Language|TLD|
+|---|---|---|---|
+|1|A|English|CA|
+|1|A|English|null|
+|1|A|null|UK|
+|1|A|English|US|
+|2|A|German|AT,DE|
+|2|B|German|CH|
+|2|B|German|null|
+
+::::{tip}
+`LOOKUP JOIN` does not guarantee the output to be in any particular order. If a certain order is required, users should use a [`SORT`](/reference/query-languages/esql/esql-commands.md#esql-sort) somewhere after the `LOOKUP JOIN`.
```

Comment:
++ this is important. We might provide _some_ guarantees later. There's an issue where the [sort order matters](https://github.com/elastic/elasticsearch/issues/121884#issuecomment-2643056468) (guarantees to be determined pre-GA).